tests/common.go

```diff
@@ -2111,19 +2114,43 @@ func AfterEachTest(contexts []*scheduler.Context, ids ...int) {
 
 // SetClusterContext sets context to clusterConfigPath
 func SetClusterContext(clusterConfigPath string) error {
+	// an empty string indicates the default kubeconfig.
+	// This variable is used to clearly indicate that in logs
+	var clusterConfigPathOrDefault = clusterConfigPath
+	if clusterConfigPathOrDefault == "" {
+		clusterConfigPathOrDefault = "default"
+	}
+	log.InfoD("Switching context to [%s]", clusterConfigPathOrDefault)
+	if clusterConfigPath == CurrentClusterConfigPath {
+		log.InfoD("Switching context: The context is already [%s]", clusterConfigPathOrDefault)
+		return nil
+	}
+
 	err := Inst().S.SetConfig(clusterConfigPath)
 	if err != nil {
-		return fmt.Errorf("Failed to switch to context. Set Config Error: [%v]", err)
+		return fmt.Errorf("failed to switch to context. Set Config Error: [%v]", err)
 	}
+
 	err = Inst().S.RefreshNodeRegistry()
 	if err != nil {
-		return fmt.Errorf("Failed to switch to context. RefreshNodeRegistry Error: [%v]", err)
+		return fmt.Errorf("failed to switch to context. RefreshNodeRegistry Error: [%v]", err)
 	}
 
 	err = Inst().V.RefreshDriverEndpoints()
 	if err != nil {
-		return fmt.Errorf("Failed to switch to context. RefreshDriverEndpoints Error: [%v]", err)
+		return fmt.Errorf("failed to switch to context. RefreshDriverEndpoints Error: [%v]", err)
 	}
+
+	if sshNodeDriver, ok := Inst().N.(*ssh.SSH); ok {
+		err = ssh.RefreshDriver(sshNodeDriver)
```

Comment:
How about sshNodeDriver.updateDriver()?

Reply:
Refreshing, in this case, is the same as "updating" (which is what I named the function). But it doesn't have to be that way, so having the name "refresh" makes more sense here, especially if more logic is introduced in `RefreshDriver`, but not in `updateDriver` in the future, since `updateDriver` is also used by `Init`. This is more consistent with the behaviour of other drivers

Reply:
To ensure that refreshDriver always uses the current implementation of code, you can consider copying and pasting the updateDriver code instead of calling it.

Reply:
I don't understand why one would duplicate the code. We know that the logic in `updateDriver` _definitely_ has to be used whenever there's initialization _or_ reinit (aka refresh), so it makes sense to call it from both functions, not duplicate the code.

Reply:
Makes sense.